EventLog.md

```diff
@@ -0,0 +1,397 @@
+  case class EventContinueObserve(timestamp: Instant, sid: Sequence.Id, why: Option[String]) extends Event
+  case class EventAbortObserve(   timestamp: Instant, sid: Sequence.Id, why: Option[String]) extends Event
+  case class EventStopObserve(    timestamp: Instant, sid: Sequence.Id, why: Option[String]) extends Event
+
+  case class EventStartIntegration(timestamp: Instant, sid: Sequence.Id, step: Int) extends Event
+  case class EventEndIntegration(  timestamp: Instant, sid: Sequence.Id, step: Int) extends Event
+}
+````
+
+Here I replaced "dataset" start/end events with "integration" start/end keeping in mind that multiple datasets may be produced for each step.
+
+As indicated by vertical whitespace in the code above there are three categories of events.  Correspondingly, I assume there will need to be three tables to store them (?).
+
+### 1. Observe Stage Events
+
+Stages of observation execution: slew, visit, and sequence.  Visit is a concept that I'm not sure I understand.  Is there a difference between visit start/end and sequence start/end?  A couple of `ENUM` types will be necessary to define the table.  First, the event lifecycle to mark the start and end events:
+
+````sql
+CREATE TYPE evt_lifecycle AS ENUM (
+    'Start',
+    'End'
+);
+
+ALTER TYPE evt_lifecycle OWNER TO postgres;
+````
+
+Which are translated into generated Scala enum types in `gen2`:
+
+````scala
+      enum("EventLifecycle") {
+        type EventLifecycleRec = Record.`'tag -> String`.T
+        val io = sql"""
+          SELECT enumlabel a, enumlabel b
+          FROM pg_enum JOIN pg_type ON pg_enum.enumtypid = pg_type.oid
+          WHERE pg_type.typname = 'evt_lifecycle'
+        """.query[(String, EventLifecycleRec)].list
+        io.transact(xa).unsafePerformIO
+      }
+````
+
+Second, the stage type itself:
+
+````sql
+CREATE TYPE evt_observe_stage AS ENUM (
+    'Sequence',
+    'Slew',
+    'Visit'
+);
+
+ALTER TYPE evt_observe_stage OWNER TO postgres;
+````
+
+with corresponding code to generate a Scala `EventObserveStage` enum.  Finally, the table itself:
+
+````sql
+CREATE TABLE log_observe_stage
+(
+   id            SERIAL,
+   "timestamp"   timestamp (5) WITH TIME ZONE NOT NULL DEFAULT CURRENT_TIMESTAMP,
+   lifecycle     evt_lifecycle NOT NULL,
+   observe_stage evt_observe_stage NOT NULL,
+   sequence_id   text NOT NULL
+);
+ 
+ALTER TABLE log_observe_stage OWNER TO postgres;
+
+ALTER TABLE ONLY log_observe_stage
+    ADD CONSTRAINT log_observe_stage_pkey PRIMARY KEY (id);
+
+CREATE INDEX ix_log_observe_stage_timestamp ON log_observe_stage USING btree ("timestamp" DESC);
+````
+
+There are a couple of things I'm not sure about here.  First, `timestamp`, even in combination with sequence id, doesn't seem like a valid primary key.  Do we need a primary key at all?  We'll never actually join on the primary key, I think. The reason I could find for it was avoiding duplicates when something goes wrong reading from a backup.
+
```

Comment:
I'd think is always better that the primary key has no real meaning, it is just an id